ddtrace/contrib/botocore/services/sqs.py

```diff
@@ -0,0 +1,228 @@
+import json
+from typing import Any  # noqa:F401
+from typing import Dict  # noqa:F401
+from typing import Optional  # noqa:F401
+
+import botocore.client
+import botocore.exceptions
+
+from ddtrace import Span  # noqa:F401
+from ddtrace import config
+from ddtrace.internal.schema.span_attribute_schema import SpanDirection
+
+from ....ext import SpanTypes
+from ....ext import http
+from ....internal.datastreams.processor import PROPAGATION_KEY_BASE_64
+from ....internal.logger import get_logger
+from ....internal.schema import schematize_cloud_messaging_operation
+from ....internal.schema import schematize_service_name
+from ....pin import Pin  # noqa:F401
+from ....propagation.http import HTTPPropagator
+from ..utils import extract_trace_context_json
+from ..utils import get_pathway
+from ..utils import get_queue_name
+from ..utils import set_patched_api_call_span_tags
+from ..utils import set_response_metadata_tags
+
+
+log = get_logger(__name__)
+
+
+def _encode_data(trace_data):
+    """
+    This method exists solely to enable us to patch the value in tests, since
+    moto doesn't support auto-encoded SNS -> SQS as binary with RawDelivery enabled
+    """
+    return json.dumps(trace_data)
+
+
+def get_topic_arn(params):
+    # type: (str) -> str
+    """
+    :params: contains the params for the current botocore action
+    Return the name of the topic given the params
+    """
+    sns_arn = params["TopicArn"]
+    return sns_arn
+
+
+def inject_trace_data_to_message_attributes(trace_data, entry, endpoint_service=None):
+    # type: (Dict[str, str], Dict[str, Any], Optional[str]) -> None
+    """
+    :trace_data: trace headers and DSM pathway to be stored in the entry's MessageAttributes
+    :entry: an SQS or SNS record
+    :endpoint_service: endpoint of message, "sqs" or "sns"
+    Inject trace headers and DSM info into the SQS or SNS record's MessageAttributes
+    """
+    if "MessageAttributes" not in entry:
+        entry["MessageAttributes"] = {}
+    # Max of 10 message attributes.
+    if len(entry["MessageAttributes"]) < 10:
+        if endpoint_service == "sqs":
+            # Use String since changing this to Binary would be a breaking
+            # change as other tracers expect this to be a String.
+            entry["MessageAttributes"]["_datadog"] = {"DataType": "String", "StringValue": _encode_data(trace_data)}
+        elif endpoint_service == "sns":
+            # Use Binary since SNS subscription filter policies fail silently
+            # with JSON strings https://github.com/DataDog/datadog-lambda-js/pull/269
+            # AWS will encode our value if it sees "Binary"
+            entry["MessageAttributes"]["_datadog"] = {"DataType": "Binary", "BinaryValue": _encode_data(trace_data)}
+        else:
+            log.warning("skipping trace injection, endpoint is not SNS or SQS")
```

Comment:
Can these logs be noisy? Should we make them debug logs or is it a sign something is very wrong